scijava-ops-image/src/main/java/org/scijava/ops/image/threshold/AbstractApplyThresholdIterable.java

```diff
@@ -39,11 +40,11 @@
  * @author Christian Dietz (University of Konstanz)
  */
 public abstract class AbstractApplyThresholdIterable<T> implements
-	Computers.Arity1<Iterable<T>, Iterable<BitType>>
+	Computers.Arity1<RandomAccessibleInterval<T>, RandomAccessibleInterval<BitType>>
```

Comment:
Why does the change from `Iterable` to `RAI` enable better efficiency?

Does the fact that `RAI` implements `IterableInterval` in ImgLib2 v7 make a difference in our decision here?

Reply:
> Why does the change from `Iterable` to `RAI` enable better efficiency?

Well, I haven't tested, so better efficiency is probably a stretch to say, however [this dependency](https://github.com/scijava/scijava/pull/266/files#diff-a338c8eaa348b8bc4b9660eac526fae6fd82acb330ec04f98559b41e3e3162eeR47) can now make use of `LoopBuilder`, which is nice.

> Does the fact that `RAI` implements `IterableInterval` in ImgLib2 v7 make a difference in our decision here?

Probably not? There are likely some changes (like [this](https://github.com/scijava/scijava/pull/266/files#diff-c6e178963f34df81fa66723a93cdd7f95ba45232e7cd804b7b3bab2dee2e3a40R56)) that could be reverted, but I can't imagine this would break?

Reply:
> I can't imagine this would break?

I wasn't suggesting that this change, together with the ImgLib2 v7 update, would break anything. Rather, I'm wondering if the ImgLib2 v7 update would somehow obviate the need to narrow the scope of the threshold ops from `Iterable` to `RAI`.

> [this dependency](https://github.com/scijava/scijava/pull/266/files#diff-a338c8eaa348b8bc4b9660eac526fae6fd82acb330ec04f98559b41e3e3162eeR47) can now make use of LoopBuilder, which is nice.

Yep, that's good. But for `Iterable`s generally, there is the `spliterator()` subsystem, which could potentially enable multithreaded thresholding over any `Iterable`. Though I suppose the "proxy" behavior of `T` instances might preclude the ability to loop in parallel that way...

Another option would be to keep the existing thresholding Ops for `Iterable` and simply add the RAI-specific ones at higher priority. Then we wouldn't need to bump scijava-ops-image to 2.0.0 in response to the backwards incompatibility.